### ISSUE TITLE
In-Game: "other" behaviour changes in 2024.1100 appear completely broken with chained static methods

### ISSUE THREAD

**Issue by thennothinghappened** [YoYoGames/GameMaker-Bugs#8251]
### Description
The way `other` works when methods are chained with different objects seems to be broken, as I can't imagine this is intended to work as it does. Depending on if the methods are `static` also affects the behaviour.

Legacy behaviour seems to work as I'd expect.

### Steps To Reproduce
See attached project (or copy GML below) and note the debug logs, comparing this against the legacy behaviour.

If methods are `static`, then `other` never changes, and always shows as `Object`. If methods are made not static, then `other` becomes a mirror of `self`. Neither behaviour seems to be correct.

#### Legacy Behaviour (Reference)
```haskell
Controller::doThing - self: Controller, other: Object
Node::bounceBack - self: Node, other: Controller
Controller::callThisFromNode - self: Controller, other: Node
```

#### `static method = ...`
```haskell
Controller::doThing - self: Controller, other: Object
Node::bounceBack - self: Node, other: Object
Controller::callThisFromNode - self: Controller, other: Object
```

#### `method = ...`
```haskell
Controller::doThing - self: Controller, other: Controller
Node::bounceBack - self: Node, other: Node
Controller::callThisFromNode - self: Controller, other: Controller
```

```gml
function Controller(node) constructor {
	
	self.node = node;
	self.node.controller = self;
	
	static doThing = function() {
		show_debug_message($"Controller::doThing - self: {instanceof(self)}, other: {instanceof(other)}");
		self.node.bounceBack();
	};
	
	static callThisFromNode = function() {
		show_debug_message($"Controller::callThisFromNode - self: {instanceof(self)}, other: {instanceof(other)}");
	};
	
}

function Node() constructor {
	
	static bounceBack = function() {
		show_debug_message($"Node::bounceBack - self: {instanceof(self)}, other: {instanceof(other)}");
		self.controller.callThisFromNode();
	};
	
}

var controller = new Controller(new Node());
controller.doThing();
```

### Which version of GameMaker are you reporting this issue for?
IDE v2024.1100.0.665 Runtime v2024.1100.0.691

### Which operating system(s) are you seeing the problem on?
Windows 10.0.19045.0

### Which platform(s) are you seeing the problem on?
Windows

[3412f81b-6bec-42b7-b6cf-45e8f137f276](https://api.gamemaker.io/api/github/downloads/3412f81b-6bec-42b7-b6cf-45e8f137f276)

**Comment by rwkay** [comment]
Fixed in 2024.11

**Comment by mgeddesGM** [comment]
verified as of  IDE v2024.1100.0.671  Runtime v2024.1100.0.696
